test/extensions/transport_sockets/starttls/starttls_integration_test.cc

```diff
@@ -126,31 +127,6 @@ class StartTlsSwitchFilterConfigFactory : public Extensions::NetworkFilters::Com
   const std::string name_;
 };
 
-// ClientTestConnection is used for simulating a client
-// which initiates a connection to Envoy in clear-text and then switches to TLS
-// without closing the socket.
-class ClientTestConnection : public Network::ClientConnectionImpl {
```

Comment:
I think that you should put this class back into the test. I see below that you use `Network::ClientConnectionImpl` and use `StartTls` as transport socket there. This effectively tests `StartTls` vs `StartTls`. So imagine that there is a bug in `startSecureTransport` which does not switch to TLS, but keeps sending in cleartext. Integration tests will pass because the same bug is on sending and receiving end. To protect again such cases, `StartTls` should only be on one end of the connection.